src/main/java/org/dependencytrack/resources/v1/ConfigPropertyResource.java

```diff
@@ -156,5 +160,26 @@ public Response updateConfigProperty(List<ConfigProperty> list) {
         return Response.ok(returnList).build();
     }
 
-
+    @GET
+    @Path("/public/{groupName}/{propertyName}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Returns a public ConfigProperty", description = "<p></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Public ConfigProperty returned", content = @Content(schema = @Schema(implementation = ConfigProperty.class))),
+            @ApiResponse(responseCode = "403", description = "This is not a public visible ConfigProperty")
+    })
+    @AuthenticationNotRequired
+    public Response getGreetingMessage(
+            @Parameter(description = "The group name of the value to retrieve", required = true) @PathParam("groupName") String groupName,
+            @Parameter(description = "The property name of the value to retrieve", required = true) @PathParam("propertyName") String propertyName) {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            ConfigProperty property = qm.getConfigProperty(groupName, propertyName);
+            qm.close();
+            ConfigPropertyConstants publicConfigProperty = ConfigPropertyConstants.ofProperty(property);
+            if (!publicConfigProperty.getIsPublic()) {
+                return Response.status(Response.Status.FORBIDDEN).entity("This is not a public visible ConfigProperty").build();
+            }
```

Comment:
A few nits:

* You don't need to close `qm` manually - that's what the try-with-resources is for
* Consider performing the `isPublic` validation *before* querying the database, potentially even before you open the `QueryManager` - database access is not required for the validation